### PR TITLE
Make CI job names more readable

### PR DESCRIPTION
**Pull request by dmitriy-sobolev**
Change mixed approach for naming jobs to "name=value,name_1=value_1,...,name_n=value_n" for values which need keys for clarity. 

Name before the changes:
`ubuntu-20.04_g++_backend=tbb_device=HOST_cxx17_release`
Name after the changes:
`ubuntu-20.04,cmplr=g++,backend=tbb,device=HOST,std=17,build_type=release`

Upd. The most important properties are shown first due to 28-symbol limit of the view panel:
`HOST,bknd=tbb,cmplr=g++,ubuntu-20.04,std=c++17,cfg=release`

Previous '_' separator might have confused c++ developers. For example, ones may get surprised to find 'tbb_device'.

### CODE REVIEW

**Review by rarutyun — COMMENTED** [review]
I like that change, but I would like to see what others are thinking.

This is a huge improvement to me. Thanks.

**Review by danhoeflinger — COMMENTED** [review]
I also like this a lot.
The only downside I notice is that the names start to be clipped by the space allowed in github for checks.  
If we could shave a few characters without confusing matters that may be for the better.  Possible candidates which I think wouldn't cause too much confusion:
drop `os=` 
`compiler` -> `cmplr`

That would save us 6 characters and remove all of the relevant clipping of names for the current set.

**Review by danhoeflinger — APPROVED** [review]
LGTM, but you may want to get others' opinions still.

**Comment by akukanov** [comment]
Is this the order we prefer to see? Given that the view panel may truncate a lot (for example, I see only `windows-2019_dpcpp_backend=...`), what would be the most important information to show? I would say it's the backend, the device, and the compiler (in that order, but any other order would still be better than using most of the space for OS name & version)

**Comment by dmitriy-sobolev** [comment]
> Is this the order we prefer to see? Given that the view panel may truncate a lot (for example, I see only `windows-2019_dpcpp_backend=...`), what would be the most important information to show? I would say it's the backend, the device, and the compiler (in that order, but any other order would still be better than using most of the space for OS name & version)

Done. I also removed `backend=` therefore backend, device and compiler can fit 28 symbol limit of the view panel.

**Comment by akukanov** [comment]
This seems good to me, but I guess more opinions need to be collected.

**Review by SergeyKopienko — DISMISSED** [review]
LGTM

**Review by danhoeflinger — DISMISSED** [review]
Yep, I like it.  LGTM

**Review by mmichel11 — DISMISSED** [review]
LGTM.

These changes make the CI job names a lot more readable and easier to search for certain configurations.

**Comment by danhoeflinger** [comment]
FYI, I believe clang-format is complaining because the branch is not rebased with current head of main, files which are different from this branch to the current head of main have suggested changes.  No action needs to be done here before merging.

Clang-format PR #715 is still awaiting approval.  However, I believe this check would still appear red even after that is merged if any of the suggestions are in the branch to branch diff delta lines.  It should be possible to change #715 to compare to the  nearest ancestor or merge base.  We can discuss that question there...

File: .github/workflows/ci.yml
Comment (rarutyun) [inline .github/workflows/ci.yml:75]: I slightly prefer to have `backend` instead of `be`. If people think otherwise, I am ready to approve it as well. Overall - big improvement. Please also rebase it on the last state of the main branch as Dan suggested

Comment (dmitriy-sobolev) [reply .github/workflows/ci.yml:77]: I do not have strong opinion on that as well.
@danhoeflinger what do you think?

Comment (danhoeflinger) [reply .github/workflows/ci.yml:77]: I think `be` is clear enough, especially after people get used to it.  I think characters come at a high cost here, so I'd keep it as it is if it were up to me.

**Review by julianmi — COMMENTED** [review]
I like these changes a lot. In case we need to reduce the number of characters, I think we could drop `os=` since the OS strings are descriptive in my opinion and the `=` in `std`.

Why don't we report the device for macOS for consistency?

**Comment by dmitriy-sobolev** [comment]
> I like these changes a lot. In case we need to reduce the number of characters, I think we could drop `os=` since the OS strings are descriptive in my opinion and the `=` in `std`.
> 
> Why don't we report the device for macOS for consistency?

I think the further reduction is no more relevant because: 
 - Everything already fits in "Show all checks panel" in the "Conversation" panel. 
 - Job names are limited to 28 symbols in the "Checks" tab and it is possible to show up to the first 3 arguments. It is possible to fit 4th argument sacrificing keys. I think it does not worth so. 

MacOS does not support CPU and GPU offloading so far, so the implied device is always HOST.

File: .github/workflows/ci.yml
Comment (rarutyun) [reply .github/workflows/ci.yml:77]: @danhoeflinger It took me several to figure it out what be means (I had several variants). But I don't want to insist

**Review by rarutyun — APPROVED** [review]

**Comment by julianmi** [comment]
> MacOS does not support CPU and GPU offloading so far, so the implied device is always HOST.

So, should we include the device anyways to make it more explicit?

**Comment by dmitriy-sobolev** [comment]
> So, should we include the device anyways to make it more explicit?

It's done.

File: .github/workflows/ci.yml
Comment (danhoeflinger) [reply .github/workflows/ci.yml:77]: Is there any *visible* place we can place a legend for the checks?  I'm not seeing a great place, but I don't know much about options available to admins for the repo.
I don't have a strong opinion here, `bknd` is a more explicit middleground but still kicks out the compiler from the "checks" view.

Comment (dmitriy-sobolev) [reply .github/workflows/ci.yml:77]: I checked available tools which I have for my own forks. Looks like there are no such options.

I've seen another approach which solves that problem: create multiple workflow files, thus moving one parameter to a workflow name. This could be a backend for oneDPL. This will result in the following:

-  "Checks" tab will have multiple workflows
- "Show all checks" panel in the "Conversation" tab will list jobs like

```
dpcpp backend / device=CPU,cmplr=icpx,os=ubuntu-20.04,std=17,cfg=release
...
tbb backend / cmplr=g++,os=ubuntu-18.04,std=17,cfg=release
...
```
I think the current readability is not that bad for that change...

Comment (danhoeflinger) [reply .github/workflows/ci.yml:77]: Thanks for looking at that.  I would vote to merge as it is and if it creates confusion, we could explore something like that in the future.

**Comment by akukanov** [comment]
One more suggestion to try: how about moving device to the first place, and dropping the prefix? It will also give some space to change backend prefix to `bknd`.
It would look like this: `HOST,bknd=serial,cmplr=g++,os=ubuntu-20.04,std=17,cfg=release` and when truncated `HOST,bknd=serial,cmplr=g++,o...`

File: .github/workflows/ci.yml
Comment (akukanov) [inline .github/workflows/ci.yml:77]: If the overall length is not a limit, I suggest to change to `std=c++${{ matrix.std }}` (e.g. `std=c++17`) so that it is similar to the well-known compiler option.

Otherwise, the improvements look very good to me.

**Review by akukanov — APPROVED** [review]

**Pull request merged**
